source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1175 @@
 #include <cmath>
 #include <limits>
 
+
 DEAL_II_NAMESPACE_OPEN
 
 
 namespace GridGenerator
 {
+  namespace Airfoil
+  {
+    AdditionalData::AdditionalData()
+      // airfoil configuration
+      : airfoil_type("NACA")
+      , naca_id("2412")
+      , center(-0.1, 0.14)
+      , airfoil_length(1.0)
+      // far field
+      , height(30.0)
+      , length_b2(15.0)
+      // mesh
+      , incline_factor(0.35)
+      , bias_factor(2.5)
+      , refinements(2)
+      , n_subdivision_x_0(3)
+      , n_subdivision_x_1(2)
+      , n_subdivision_x_2(5)
+      , n_subdivision_y(3)
+      , point_factor(2)
+    {
+      Assert(
+        airfoil_length <= height,
+        ExcMessage(
+          "Mesh is to small to enclose airfoil! Choose larger field or smaller chord length!"));
+    }
+
+
+
+    void
+    AdditionalData::declaration(ParameterHandler &prm)
+    {
+      prm.enter_subsection("FarField");
+      {
+        prm.declare_entry("Height",
+                          std::to_string(height),
+                          Patterns::Double(),
+                          "CoarseGridBlockHight");
```

Comment:
For example, this would become:

`prm.add_entry("Height", height, "CoarseGridBlockHight");`

and you could remove the corresponding entry in the `get` function.

Reply:
I removed function get() in AdditionalData as you suggested and used add_parameter() to attach the parameters instead. 
Somehow, the function `create_triangulation(tria)` and `create_triangulation(tria, periodic_faces)` which should produce a default mesh since no `AdditionalData data` object is passed as input, doesn't work without the inputfile (parse_input(inputfile)).

Is it possible that this change (using add_paramter instead declare_entry() and get()) caused that issue?

Reply:
This is totally unrelated. My guess is that the default object you create is invalid (i.e., maybe you forgot to initialize some of the members of AdditionalData), and they are initialized by the add_parameters method instead of being initialized at construction time. 

I'd move the initialization where you declare the objects, i.e.,

`
center = Point<2>(0,0);
`
instead of having them initialized in the constructor. This documents the default values in the same place where you declare the members, and ensures that no entries are left un-initialized.

The add_parameter method simply sets the values when it is called, or when a file is parsed.

Reply:
The mentioned problem was caused, as Peter expected, by a mistake in my test program.

Should I still move the initialization of `center` as you suggested?

Reply:
@luca-heltai Initializing default values in the initializer list is how we normally do it, isn't it?